src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -682,7 +672,6 @@ public static boolean hasCustomDataPath(@IndexSettings Settings indexSettings) {
      * @param indexSettings settings for the index
      */
     private Path resolveCustomLocation(@IndexSettings Settings indexSettings) {
-        assert indexSettings != ImmutableSettings.EMPTY;
```

Comment:
Hmm can we put this back once you address the nocommit about getting the right settings passed down?